src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -48,6 +48,88 @@ internal abstract class IpcEndpoint
         public abstract Task WaitForConnectionAsync(CancellationToken token);
     }
 
+    internal class IpcEndpointConfig
+    {
+        internal enum PortType
+        {
+            None,
+            Connect,
+            Listen
+        }
+
+        public string Address { get; } = "";
+        public PortType Type { get; } = PortType.None;
+
+        public IpcEndpointConfig(string config)
```

Comment:
The None value was added to make it possible to test that you don't have a valid address, but in theory you can do the same checking for null or empty address property but felt less obvious, maybe it would be better to have properties to check what type it has instead of accessing the underlying enum, like isListen and isConnect and in case when we have address null or empty string both will return false. I write something up.